commit 7ed7575b2b25ba9f3d1eba186facd9e70f928c1e (refs/changes/97/5297/1)
Author: Peter Marton <pmarton@netflix.com>
Date:   2019-01-02T11:38:25-08:00 (9 months ago)
    
    refactor: do not call sprintf with zero argument, performance optimization

diff --git a/lib/verror.js b/lib/verror.js
index 8663dde..45e4ac1 100644
--- a/lib/verror.js
+++ b/lib/verror.js
@@ -114,8 +114,13 @@ function parseConstructorArguments(args)
 		});
 	}
 
+	/**
+	 * Do not call sprintf with one argument, performance optimization
+	 */
 	if (sprintf_args.length === 0) {
 		shortmessage = '';
+	} else if (sprintf_args.length === 1) {
+		shortmessage = sprintf_args[0];
 	} else {
 		shortmessage = sprintf.apply(null, sprintf_args);
 	}
